src/main/java/org/jitsi/jicofo/xmpp/IqHandler.java

```diff
@@ -334,31 +336,57 @@ private IQ handleRayoIQ(RayoIqProvider.DialIq dialIq, int retryCount, @NotNull L
 
         try
         {
-            IQ reply = connection.sendPacketAndGetReply(forwardDialIq);
-
-            if (reply == null)
-            {
-                if (retryCount > 0)
+            connection.sendIqWithResponseCallback(
+                forwardDialIq,
+                reply ->
                 {
-                    exclude.add(jigasiJid);
-
-                    // let's retry lowering the number of attempts
-                    return this.handleRayoIQ(dialIq, retryCount - 1, exclude);
-                }
-                else
+                    // Send Jigasi response back to the client
+                    reply.setFrom((Jid)null);
+                    reply.setTo(from);
+                    reply.setStanzaId(dialIq.getStanzaId());
+
+                    connection.sendStanza(reply);
+                },
+                exception ->
                 {
-                    return IQ.createErrorResponse(
-                        dialIq, XMPPError.getBuilder(XMPPError.Condition.remote_server_timeout));
-                }
-            }
+                    logger.error("Error sending dialIQ to "+ jigasiJid + " " + exception.getMessage());
+
+                    try
+                    {
+                        if (retryCount > 0)
+                        {
+                            excludeList.add(jigasiJid);
+
+                            // let's retry lowering the number of attempts
+                            IQ result = this.handleRayoIQ(dialIq, retryCount - 1, excludeList);
```

Comment:
Looks like `this.handleRayoIQ` now unconditionally returns `null`

Reply:
It can return few errors.